src/platform/tasks/vscode/tasksService.ts

```diff
@@ -42,6 +57,19 @@ export class TasksService implements ITasksService {
 		return existingTasks.some(t => this.matchesTask(t, def));
 	}
 
+	/**
+	 * This is needed because when tasks exit, they're removed from the taskExecutions, but we might want to review the output of the task
```

Comment:
here's how that keyedTaskId is created 
https://github.com/microsoft/vscode/blob/0748e889c18201cf0040c9022e5f87d4aaabf249/src/vs/workbench/contrib/tasks/common/tasks.ts#L1296-L1318